### PR TITLE
GlobalExceptionHandler and data validation

### PR DESCRIPTION
**Pull request by 4Vitalii5**
Requirements
Add CustomGlobalExceptionHandler class
Add meaningful validation for existing DTO classes

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,47 +37,43 @@
...
-
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
             <optional>true</optional>
         </dependency>
-
+        <dependency>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
```
Comment (Elena-Bruyako) [inline pom.xml]: it's better to use spring-boot-starter-validation

File: src/main/java/mate/academy/dto/BookDto.java [hunk 2]
```diff
@@ -1,7 +1,25 @@
...
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
+import mate.academy.validation.Isbn;
 
-public record BookDto(Long id, String title, String author, String isbn, BigDecimal price,
-                      String description, String coverImage) {
+public record BookDto(
+        @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/BookDto.java]: don't need to validate response

File: src/main/java/mate/academy/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
 
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
+import mate.academy.validation.Isbn;
 
-public record CreateBookRequestDto(String title, String author, String isbn, BigDecimal price,
-                                   String description, String coverImage) {
+public record CreateBookRequestDto(
+        @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/CreateBookRequestDto.java]:
```suggestion
        @NotBlank
```

for String

File: src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(
+            EntityNotFoundException exception, WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.NO_CONTENT);
+        body.put("message", exception.getMessage());
+        body.put("path", request.getDescription(false));
+        return new ResponseEntity<>(body, HttpStatus.NO_CONTENT);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
```

File: src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(
+            EntityNotFoundException exception, WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.NO_CONTENT);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        body.put("status", HttpStatus.NOT_FOUND);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(
+            EntityNotFoundException exception, WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.NOT_FOUND);
+        body.put("message", exception.getMessage());
+        body.put("path", request.getDescription(false));
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java]: you can move this logic to the private method and reuse for other methods ;)

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! 🔥

File: src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java (hunk 6 above)
Comment (4Vitalii5) [reply src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java]: Thank you.

**Pull request merged**